src/inmanta/agent/io/remote.py

```diff
@@ -125,10 +129,19 @@ def _build_connect_string(self) -> str:
 
     def _execute(self, function_name, *args, **kwargs):
         with self._lock:
-            ch = self._gw.remote_exec(local)
-            ch.send((function_name, args, kwargs))
-            result = ch.receive()
-            ch.close()
+            try:
+                ch = self._gw.remote_exec(local)
+            except OSError:
+                raise ChannelClosedException(
+                    "Failed to execute remote command on %(uri)s, this is most likely caused by "
```

Comment:
Let's use and f-string to replace the uri parameter.

Reply:
resolved